tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go

```diff
@@ -0,0 +1,219 @@
+		Expect(nmStateConfigList).To(BeEmpty(), "Found spoke NM state config manifests but expected to be removed")
+
+		By("checking the klusterlet addon config manifests removed on hub")
+		_, err = ocm.PullKAC(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).To(HaveOccurred(), "Found spoke kac manifests but expected to be removed")
+
+		By("checking the agent cluster install manifests removed on hub")
+		_, err = assisted.PullAgentClusterInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).To(HaveOccurred(), "Found spoke ACI manifests but expected to be removed")
+
+		By("verifying installed spoke cluster should still be functional")
+		_, err = version.GetOCPVersion(Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get OCP version from spoke and verify spoke cluster access")
+
+		// Test teardown.
+		// Test step 1-Recreate default assisted installer template reference ConfigMap CR. I.e. the one recently deleted.
+
+		// Test step 2-Recreate the ClusterInstance custom resource. I.e. the one recently deleted.
+		By("resetting the clusters app back to the original settings")
+		err = gitdetails.SetGitDetailsInArgoCd(
+			tsparams.ArgoCdClustersAppName, tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName], true, false)
+		Expect(err).ToNot(HaveOccurred(), "Failed to reset clusters app git details")
+
+		// Test step 2 expected results validation.
+		By("checking the infra env manifests exists on hub")
+		_, err = assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke infra env manifests")
+
+		By("checking the bare metal host manifests exists on hub")
+		_, err = bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke bmh manifests")
+
+		By("checking the cluster deployment manifests exists on hub")
+		_, err = hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke cluster deployment manifests")
+
+		By("checking the NM state config manifests exists on hub")
+		nmStateConfigList, err = assisted.ListNmStateConfigs(HubAPIClient, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to list NM state config manifests")
+		Expect(nmStateConfigList).ToNot(BeEmpty(), "Failed to find NM state config manifests")
+
+		By("checking the klusterlet addon config manifests exists on hub")
+		_, err = ocm.PullKAC(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke kac manifests")
+
+		By("checking the agent cluster install manifests exists on hub")
+		_, err = assisted.PullAgentClusterInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke agent cluster install manifests")
```

Comment:
are these the same between the test cases? maybe they should go into an AfterEach

Reply:
Okay, updated. Thanks!